### PR TITLE
Use BufReader and BufWriter

### PR DESCRIPTION
**Pull request by tomaka**
Fixes all the performances issues, and should also fix most of the weird disconnects.

### CODE REVIEW

File: substrate/network-libp2p/src/topology.rs [hunk 1]
```diff
@@ -101,7 +101,7 @@ impl NetTopology {
 		};
 
 		let file = fs::File::create(path)?;
-		serialize(file, &self.store)
+		serialize(BufWriter::with_capacity(1024 * 1024, file), &self.store)
```
[the remark this answers is not in the document]
Comment (tomaka) [reply substrate/network-libp2p/src/topology.rs]: Considering that the file is 45MB right now, the default of 8kB looked a bit too low to me.

[the remark this answers is not in the document]
Comment (dvdplm) [reply substrate/network-libp2p/src/topology.rs]: So why 1? Why not 2? Or 5 or 45?

[the remark this answers is not in the document]
Comment (tomaka) [reply substrate/network-libp2p/src/topology.rs]: Well, an arbitrary had to be set.
Ideally we would benchmark multiple values, but this doesn't seem very productive to me at this stage.

[the remark this answers is not in the document]
Comment (pepyakin) [reply substrate/network-libp2p/src/topology.rs]: agree with @tomaka , TODO wouldn't hurt though

**Pull request merged**